File: test-selection-mcp/vscode-test-workspace/test-code.py
# Python test code for selection injection testing
def calculate_fibonacci(n):
    """Calculate fibonacci number using recursion."""
    if n <= 1:
        return n
    return calculate_fibonacci(n-1) + calculate_fibonacci(n-2)

class DataProcessor:
    def __init__(self, data):
        self.data = data
    
    def process(self):
        return [x * 2 for x in self.data if x > 0]

# Test data
numbers = [1, -2, 3, -4, 5]
processor = DataProcessor(numbers)
result = processor.process()
print(f"Fibonacci of 10: {calculate_fibonacci(10)}")
print(f"Processed data: {result}")

File: test-selection-mcp/vscode-test-workspace/test-code.js
// JavaScript test code for selection injection testing
function calculateSum(arr) {
    return arr.reduce((sum, num) => sum + num, 0);
}

async function fetchUserData(userId) {
    try {
        const response = await fetch(`/api/users/${userId}`);
        const userData = await response.json();
        return userData;
    } catch (error) {
        console.error('Failed to fetch user data:', error);
        throw error;
    }
}

class TaskManager {
    constructor() {
        this.tasks = [];
    }
    
    addTask(task) {
        this.tasks.push({ ...task, id: Date.now(), completed: false });
    }
    
    completeTask(taskId) {
        const task = this.tasks.find(t => t.id === taskId);
        if (task) task.completed = true;
    }
}

// Usage example
const numbers = [1, 2, 3, 4, 5];
const sum = calculateSum(numbers);
console.log(`Sum: ${sum}`);
